staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -196,14 +204,52 @@ func (a *GenericAdmissionWebhook) Admit(attr admission.Attributes) error {
 	hooks := hookConfig.ExternalAdmissionHooks
 	ctx := context.TODO()
 
-	errCh := make(chan error, len(hooks))
+	hookIndexCh := make(chan int, len(hooks))
 	wg := sync.WaitGroup{}
 	wg.Add(len(hooks))
 	for i := range hooks {
+		go func(hook *v1alpha1.ExternalAdmissionHook, i int) {
+			defer wg.Done()
+			if shouldCallHook(hook, attr) {
+				hookIndexCh <- i
+			}
+		}(&hooks[i], i)
+	}
+	wg.Wait()
+	close(hookIndexCh)
+
+	var hookIndice []int
+	for index := range hookIndexCh {
+		hookIndice = append(hookIndice, index)
+	}
+	if len(hookIndice) == 0 {
+		// no matching hooks
+		return nil
+	}
+
+	// convert the object to the external version before sending it to the webhook
+	var versioned versionedObjects
+	gvr := attr.GetResource()
+	if attr.GetOldObject() != nil {
+		versioned.oldObject, err = a.convertor.ConvertToVersion(attr.GetOldObject(), schema.GroupVersion{Group: gvr.Group, Version: gvr.Version})
+		if err != nil {
+			return err
+		}
+	}
+	if attr.GetObject() != nil {
+		versioned.object, err = a.convertor.ConvertToVersion(attr.GetObject(), schema.GroupVersion{Group: gvr.Group, Version: gvr.Version})
+		if err != nil {
+			return err
+		}
+	}
+
+	errCh := make(chan error, len(hookIndice))
+	wg.Add(len(hookIndice))
+	for i := range hookIndice {
```

Comment:
`_, i` and then you can omit `hookIndice` below. Or switch to ranging over the channel.

Reply:
Right. Anyway , by removing the parallel code, this is gone.